Refactor reflected regions background estimation

joleroi opened the pull request:
This PR

* removes the function  ``gammapy.background.reflected_regions_background_estimate``
* add the class ``gammapy.background.ReflectedRegionsBackgroundEstimator``
* removes the ``gammapy.data.TargetSummary`` class which was not used anywhere and bad

which leads to a nicer API.

This needs a follow up PR where the new class is properly used in all places (at the moment I changed as little as possible to not break anything)

cdeil commented:
👍  to turn this into a class.

As I mentioned, I started a branch to do this when I changed Gammapy over to `gammapy.regions`, but then never finished it. It's here:
https://github.com/cdeil/gammapy/blob/c736a38842ac8959ff42343d527a806e45843880/gammapy/background/reflected.py

Feel free to read over what I did there, and take any ideas or copy & paste code over if you think it's useful. I don't remember how far along I was or if it was working code. From a brief look, it seems useful. I think the basic idea was to implement the algorithm as a class for easy debugging of intermediate results as attributes / multi-step, but then expose it as a function that instantiates and runs the algorithm class.
I'm also OK with exposing it as a class to the end user.

Missing requires decorator on test:
https://travis-ci.org/gammapy/gammapy/jobs/163686457#L1732

Unrelated? Fluke?
https://travis-ci.org/gammapy/gammapy/jobs/163686454#L1718

joleroi commented:
>  As I mentioned, I started a branch to do this when I changed Gammapy over to gammapy.regions, but then never finished it. It's here:

What you're doing there is removing the function ``find_reflected_regions`` this function is not touched in this PR, so please continue your branch. I just added a class to find reflected regions for a certain observation the algorithm is  still in the ``find_reflected_regions`` function

joleroi commented:
We can discuss if we want the actual algorithm as part of the user-exposed ``ReflectedRegionsBackgroundEstimator`` class or in a separate function/class. I would opt for the latter!

cdeil commented:
@joleroi - The notebook tests are currently broken in master.
See this build from an unrelated PR:
https://travis-ci.org/gammapy/gammapy/jobs/164899201#L2949

Can you please fix this somehow?
- merge this PR
- revert the notebook change
- disable the spectrum notebook test for now

joleroi commented:
The travis fail is unrelated
https://travis-ci.org/gammapy/gammapy/jobs/163940024#L1575
